### PR TITLE
feat(medusa, link-modules): sales channel <> pubkey link

### PR DESCRIPTION
**Pull request by fPolic**
**What**
- introduce link for SC <> PublishableApiKeys
- add migration file
- repository changes

**Depends on**
- [x] https://github.com/medusajs/medusa/pull/5450

### CODE REVIEW

**Comment by olivermrbl** [comment]
@fPolic can I get you to update this one now? Should be the final one in this category :D

**Comment by fPolic** [comment]
@olivermrbl done :)

File: packages/medusa/src/models/publishable-api-key-sales-channel.ts [hunk 1]
```diff
@@ -1,12 +1,22 @@
-import { Entity, PrimaryColumn } from "typeorm"
+import { BeforeInsert, Column, Entity } from "typeorm"
```
Comment (olivermrbl) [inline packages/medusa/src/models/publishable-api-key-sales-channel.ts]: question: Should this follow similar convention as with the OrderSalesChannel?

File: packages/medusa/src/repositories/publishable-api-key-sales-channel.ts [hunk 2]
```diff
@@ -59,20 +60,23 @@ export const PublishableApiKeySalesChannelRepository = dataSource
      *
      * @param publishableApiKeyId - publishable key id
      * @param salesChannelIds - an array of SC ids
+     * @param isMedusaV2Enabled - are Medusa modules enabled
      */
     async addSalesChannels(
       publishableApiKeyId: string,
-      salesChannelIds: string[]
+      salesChannelIds: string[],
+      isMedusaV2Enabled?: boolean
```
Comment (olivermrbl) [inline packages/medusa/src/repositories/publishable-api-key-sales-channel.ts]: comment: Don't think feature flags should be part of the signature of the public-facing service method. Also, what is this used for?

Comment (fPolic) [reply packages/medusa/src/repositories/publishable-api-key-sales-channel.ts]: forgot to remove the argument during refactor, it's not needed anymore

File: packages/medusa/src/models/publishable-api-key-sales-channel.ts (hunk 1 above)
Comment (fPolic) [reply packages/medusa/src/models/publishable-api-key-sales-channel.ts]: agree, I updated the entity

File: .changeset/rotten-lobsters-move.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+"@medusajs/link-modules": patch
```
Comment (adrien2p) [inline .changeset/rotten-lobsters-move.md:2]: todo: this seems out of date?

File: packages/link-modules/src/definitions/publishable-api-key-sales-channel.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    {
+      serviceName: "salesChannelService",
+      relationship: {
+        serviceName: LINKS.PublishableApiKeySalesChannel,
+        isInternalService: true,
+        primaryKey: "sales_channel_id",
+        foreignKey: "id",
+        alias: "publishable_keys_link",
+        isList: true,
+      },
```
Comment (adrien2p) [inline packages/link-modules/src/definitions/publishable-api-key-sales-channel.ts]: todo: probably good to add a field alias wdyt>?

File: packages/medusa/src/joiner-configs/publishable-api-key-service.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  alias: [
+    {
+      name: "publishable_api_key",
+      args: { entity: "PublishableApiKey" },
+    },
+    {
+      name: "publishable_api_keys",
+      args: { entity: "PublishableApiKey" },
+    },
+  ],
```
Comment (adrien2p) [inline packages/medusa/src/joiner-configs/publishable-api-key-service.ts]:
```suggestion
  alias: [
    {
      name: ["publishable_api_key", "publishable_api_keys"]
      args: { entity: "PublishableApiKey" },
    },
  ],
```

wdyt?

**Review by adrien2p — APPROVED** [review]
LGTM, few nits todo

File: packages/link-modules/src/definitions/publishable-api-key-sales-channel.ts (hunk 4 above)
Comment (fPolic) [reply packages/link-modules/src/definitions/publishable-api-key-sales-channel.ts]: agree, I've added SC alias on the PublishableKey side

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**